src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,66 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunStartupBenchmark(string[] args, ProcessStartInfo psi, PreDel pre, MidDel mid, PostDel post)
```

Comment:
A few comments for this API:

1. This is not actually doing a start-up benchmark.  JitBench happens to do that, but the API you've designed is fairly high-level and can support any number of metrics and arbitrary tests - not just start-up.  I'd recommend calling this something like "RunScenario", but I'm open to other names.
2. The delegate names should be expanded to describe exactly what they're for.  PreDel should probably be called something like "SetupDelegate.  PostDel should be something like "TeardownDelegate".  MidDel should probably be something like "PostIterationDelegate".
3. The argument names should be more fully worded so that someone looking at them can know what they are for.
4. We should clarify how one knows that the run succeeded.  I'm assuming that it's "we didn't get an exception", but want to be sure.